### PR TITLE
Add hooks approach

### PR DESCRIPTION
**Pull request by ilfa**
- Added hooks approach
- Tested on android
- Added unit tests
- Check types
- Github action for build, tests, and types
- Add linter

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Development playground
+
+In `TestProject` folder you can find demo application. React Native provides not the best developer experience so please follow the steps.
+
+#### Common steps
+
+1. Run `yarn install` in the project root.
+2. Go to the `TestProject` folder and run yarn install here.
+3. Set up `apiKey` in `App.js`. (TODO: use environment variable).
```
Comment (TheUnderScorer) [inline contributing.md:14]: Is the TODO still relevant? If yes, maybe we should implement it before merging this PR?

File: contributing.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+#### Troubleshooting
+1. Try `yarn start --reset-cache`
+2. Try relaunch emulator
+3. Try to delete both `node_modules` and install dependencies again in order from [common steps](#common-steps)
+
+#### Hot reload
+React native instruments don't support `npm`/`yarn` links for packages, because of this each library update needs not only building this library but reinstalling module for demo app. Current repository have 2 ways to make development more comfortable.
+
+1. Change `"main"` section in library's `package.josn` to `"./src/index.ts"`. With this fix you don't need to build library for getting updates in `TestApp`, but don't forget to revert this change.
```
Comment (TheUnderScorer) [inline contributing.md:35]: Typo:

```suggestion
1. Change `"main"` section in library's `package.json` to `"./src/index.ts"`. With this fix you don't need to build library for getting updates in `TestApp`, but don't forget to revert this change.
```

File: package.json [hunk 3]
```diff
@@ -28,14 +32,34 @@
...
+    "rollup-plugin-license": "^2.7.0",
+    "rollup-plugin-peer-deps-external": "^2.2.4",
+    "ts-jest": "^27.1.4",
     "typescript": "^4.5.4"
   },
   "peerDependencies": {
-    "react-native": "*",
-    "react": "*"
+    "react": "*",
+    "react-native": "*"
```
Comment (TheUnderScorer) [inline package.json:63]: Is it safe to allow any version of `react-native`? Could there be a possibility that some new version of `react-native` will introduce some breaking changes that will break our package?

File: contributing.md (hunk 1 above)
Comment (ilfa) [reply contributing.md:14]: Thanks! I just forgot about it.

Comment (ilfa) [reply contributing.md]: Done!

File: package.json (hunk 3 above)
Comment (ilfa) [reply package.json:63]: It is not 100% safe, but `react-native` still hasn't major release.
Let's keep it this way and collect feedback.

**Review by TheUnderScorer — APPROVED** [review]

**Pull request merged**